tt_metal/impl/device/device.hpp

```diff
@@ -219,7 +220,7 @@ class Device {
     std::vector<std::unique_ptr<CommandQueue>> sw_command_queues_;
     std::unique_ptr<SystemMemoryManager> sysmem_manager_;
     vector<std::unique_ptr<Program, detail::ProgramDeleter>> command_queue_programs_;
-    const uint8_t num_hw_cqs_;
+    uint8_t num_hw_cqs_;
```

Comment:
Why remove const qualifier?

Reply:
It's from Nick's LLVM commit, I'm guessing because it's not actually const, or promoted to a non-const variable somewhere -- I'm not sure. Description was "additional clang-16 fixes after rebase".

Reply:
This will force the class to be move only and forces you to manually write the move constructor.  I can dig up the clang compiler error if you want.

Reply:
```
./tt_metal/impl/device/device.hpp:79:13: error: explicitly defaulted move assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
    Device& operator=(Device &&other) = default;
            ^
./tt_metal/impl/device/device.hpp:223:19: note: move assignment operator of 'Device' is implicitly deleted because field 'num_hw_cqs_' is of const-qualified type 'const uint8_t' (aka 'const unsigned char')
    const uint8_t num_hw_cqs_;
                  ^
./tt_metal/impl/device/device.hpp:79:41: note: replace 'default' with 'delete'
    Device& operator=(Device &&other) = default;
```